### DISCUSSION TITLE
Chat DB download missing user chats

### DISCUSSION THREAD

**Discussion by ghost** [open-webui/open-webui#1729] [General]
# Bug Report

## Description

**Bug Summary:**
I as the Admin downloaded the entire database and opened the .db file in SQLite viewer.
I could see all the users registered on the host, but the db did not have their chats in it when I opened it.
The chat session history is needed for audit purposes and reporting.
I set the permissions for user role to "downt allow chat deletion".
Are all the prompts and responses for all users supposed to be logged in the SQLite db?

**Steps to Reproduce:**
Downloaded the db a couple of times, but no luck finding other user role chats.
I was able to see their chats when downloading the JSON with all chats, but this is less useful vs having the chat session history in a db.

**Expected Behavior:**
I was hoping to see all the chats for all users in the db after the download.

**Actual Behavior:**
I am only seeing my own chats.

## Environment

- **Open WebUI Version:** [0.1.120]

- **Operating System:** macOS

## Reproduction Details

**Confirmation:**

- [ y] I have read and followed all the instructions provided in the README.md.
- [ y] I am on the latest version of both Open WebUI and Ollama.
- [ n/a] I have included the browser console logs.
- [ n/a] I have included the Docker container logs.

## Logs and Screenshots

**Browser Console Logs:**
[Include relevant browser console logs, if applicable]

**Docker Container Logs:**
[Include relevant Docker container logs, if applicable]

**Screenshots (if applicable):**
[Attach any relevant screenshots to help illustrate the issue]

## Installation Method

[Describe the method you used to install the project,: Docker

**Comment by justinh-rahb** [comment]
I would try copying it from the container manually to verify there's no user chats showing in it:
```bash
docker cp open-webui:/app/backend/data/webui.db .
```
Can you also share how and what query you're using to extract user chats?

**Comment by ghost** [comment]
Thanks @justinh-rahb the file copy works fine on CLI per your command above. It yields the same result as the "download the database" command from the GUI. Both methods do not show user chats other than my own. I am using SQLite Browser to view the files, I have not yet started creating SQL to query the db as I dont have the full db yet...

In the admin role, I have set the "chat delete" off so that users have not been able to delete their own chats.

Any thoughts?

![Image 4-23-24 at 11 21 PM](https://github.com/open-webui/open-webui/assets/166965973/61b58cca-a234-4c1a-bad7-26cc1c447284)

**Comment by ghost** [comment]
@justinh-rahb I am assuming that all chats from all users are always supposed to be stored in the SQLite db?

**Comment by justinh-rahb** [comment]
I see different `user_id`s listed in your screenshot, you're saying these are all you? My eyes are telling me a different story.
![Screenshot 2024-04-24 at 11 22 10 AM](https://github.com/open-webui/open-webui/assets/52832301/8bf62b71-d158-4116-b8a3-cba64c4879bc)
